src/components/HorizontalStep/HorizontalStep.tsx

```diff
@@ -0,0 +1,92 @@
+import clsx from 'clsx';
+import React from 'react';
+import styles from './HorizontalStep.module.css';
+import Icon from '../Icon';
+import Text from '../Text';
+
+export interface Props {
+  /**
+   * CSS class names that can be appended to the component.
+   */
+  className?: string;
+  /**
+   * Indicates which number the step is.
+   */
+  stepNumber: number;
+  /**
+   * Text that describes the step.
+   */
+  text: string;
+  /**
+   * Indicates the state of the step.
+   */
+  variant: 'complete' | 'active' | 'incomplete';
+}
+
+/**
+ * ```ts
+ * import {HorizontalStep} from "@chanzuckerberg/eds";
+ * ```
+ *
+ * A step sub component for the <HorizontalStepper>.
+ * Determines which icon to use and places the text next to it.
+ *
+ * Example usage:
+ *
+ * ```tsx
+ * <HorizontalStep stepNumber={1} text="Step 1"  variant="incomplete" />
+ * ```
+ */
+export const HorizontalStep = ({
+  className,
+  stepNumber,
+  text,
+  variant,
+}: Props) => {
+  /**
+   * Determines which icon to display.
+   */
+  const icon =
+    variant === 'complete' ? (
+      <Icon
+        className={styles['horizontal-step__complete-icon']}
+        name="check-circle"
+        purpose="informative"
+        size="1.5rem"
+        title={`completed step ${text}`}
+      />
+    ) : variant === 'active' ? (
+      <Text
+        aria-label={`current step ${stepNumber}: ${text}`}
+        as="span"
+        className={styles['horizontal-step__number']}
+        role="img"
```

Comment:
Felt weird about this but think it should be `role="img"` because it's like an icon (svg) but using a span element for the text and borders for the circle. This also allows adding of the `aria-label` as there are reports of screen readers not reading the `aria-label` of `<span>`.
@dierat  thoughts?